hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/SchedulePreTransactionHandlerImpl.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.schedule.impl;
+
+import static com.hedera.node.app.service.mono.Utils.asHederaKey;
+import static com.hedera.node.app.service.mono.utils.MiscUtils.*;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SCHEDULED_TRANSACTION_NOT_IN_WHITELIST;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.UNRESOLVABLE_REQUIRED_SIGNERS;
+
+import com.hedera.node.app.service.mono.exceptions.UnknownHederaFunctionality;
+import com.hedera.node.app.service.schedule.SchedulePreTransactionHandler;
+import com.hedera.node.app.spi.AccountKeyLookup;
+import com.hedera.node.app.spi.PreHandleDispatcher;
+import com.hedera.node.app.spi.meta.*;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.HederaFunctionality;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import org.apache.commons.lang3.NotImplementedException;
+
+/**
+ * A {@link SchedulePreTransactionHandler} implementation that pre-computes the required signing
+ * keys (but not the candidate signatures) for each schedule operation.
+ */
+public class SchedulePreTransactionHandlerImpl implements SchedulePreTransactionHandler {
+    private final AccountKeyLookup keyLookup;
+
+    public SchedulePreTransactionHandlerImpl(final AccountKeyLookup keyLookup) {
+        this.keyLookup = keyLookup;
+    }
+
+    @Override
+    public ScheduleTransactionMetadata preHandleCreateSchedule(
+            final TransactionBody txn,
+            final AccountID payer,
+            final PreHandleDispatcher dispatcher) {
+        final var op = txn.getScheduleCreate();
+        final var meta =
+                new ScheduleSigTransactionMetadataBuilder(keyLookup)
+                        .txnBody(txn)
+                        .payerKeyFor(payer);
+
+        if (op.hasAdminKey()) {
+            final var key = asHederaKey(op.getAdminKey());
+            key.ifPresent(meta::addToReqNonPayerKeys);
+        }
+
+        final var scheduledTxn =
+                asOrdinary(op.getScheduledTransactionBody(), txn.getTransactionID());
+
+        /* We need to always add the custom payer to the sig requirements even if it equals the to level transaction
```

Comment:
This comment is unclear.

Reply:
Is there a place I can clarify? I think it meant to always add `customPayer` key, even if it is same as the top level transaction payer.